### PR TITLE
Clean up the handling of gas (use int64_t only)

### PR DESCRIPTION
**Pull request by axic**
Part of #311.

### CODE REVIEW

**Comment by axic** [comment]
@chfast I think this should be correct now (tests are failing though due to the edge cases I guess?). Can you have a look at the logic in this PR?

File: src/eei.cpp [hunk 1]
```diff
@@ -485,7 +487,7 @@ namespace hera {
       static_assert(GasSchedule::logTopic <= 65536, "Gas cost of logTopic could lead to overflow");
       static_assert(GasSchedule::logData <= 65536, "Gas cost of logData could lead to overflow");
       // Using uint64_t to force a type issue if the underlying API changes.
-      takeInterfaceGas(GasSchedule::log + (GasSchedule::logTopic * numberOfTopics) + (GasSchedule::logData * static_cast<uint64_t>(length)));
+      takeInterfaceGas(GasSchedule::log + (GasSchedule::logTopic * numberOfTopics) + (GasSchedule::logData * static_cast<int64_t>(length)));
```
Comment (chfast) [inline src/eei.cpp]: This is safe cast, so you can use `int64_t(length)`.

File: src/eei.cpp [hunk 2]
```diff
@@ -1012,7 +1018,7 @@ namespace hera {
     // Allow 16 bits here.
     static_assert(GasSchedule::copy <= 65536, "Gas cost of copy could lead to overflow");
     // Using uint64_t to force a type issue if the underlying API changes.
-    takeInterfaceGas(GasSchedule::copy * ((static_cast<uint64_t>(length) + 31) / 32));
+    takeInterfaceGas(GasSchedule::copy * ((static_cast<int64_t>(length) + 31) / 32));
```
Comment (chfast) [inline src/eei.cpp]: This cast is safe, could use `int64_t(length)`.

File: src/eei.cpp [hunk 3]
```diff
@@ -1001,7 +1007,7 @@ namespace hera {
   /*
    * Utilities
    */
-  void EthereumInterface::safeChargeDataCopy(uint32_t length, uint32_t baseCost) {
+  void EthereumInterface::safeChargeDataCopy(uint32_t length, unsigned baseCost) {
```
Comment (chfast) [inline src/eei.cpp]: Why have you changed this?
From my little experience I recommend using `int` or `int64_t` for cost constants as well.

Comment (axic) [reply src/eei.cpp]: The constants are currently defined as `unsigned`, wanted to ensure it is the same here.

Comment (chfast) [reply src/eei.cpp]: :+1:
